### ISSUE TITLE
Ordering of versions seems incorrect

### ISSUE THREAD

**Issue by enyo** [dart-lang/pub-dartlang-dart#53]
If there is a newer version that is still a release candidate, it gets listed _below_ all "non release-candidate" versions.
For example here: https://pub.dartlang.org/packages/polymer/versions

```
0.16.4+2
0.16.4+1
0.16.4
...
0.5.0+1
0.5.0
1.0.0-rc.17
1.0.0-rc.16
1.0.0-rc.15
```

Even though it's a release candidate, shouldn't it be listed at the top?

**Comment by mkustermann** [comment]
This was implemented on purpose in f360308c8cf4c41f8124b4c80b1f28469ce6f7fa.

There are two orderings for package versions: Semantic versioning order and Pub's prioritization order. If you just depend on a pub package with an `any` constraint, then pub will pick the **latest stable** version (i.e. **not the latest** version) -- even if there is a newer pre-release version according to semantic version ordering. 

So the ordering you see in https://pub.dartlang.org/packages/polymer/versions is following pubs ordering (this was changed due to a user request, but I can't seem to find the bug at the moment). 

You can find documentation at [package:pub_semver/Version.prioritize](https://www.dartdocs.org/documentation/pub_semver/1.3.0/pub_semver/Version/prioritize.html)  about this.

Since we were **explicitly asked** to order versions that way, it doesn't seem to make sense to revert  f360308c8cf4c41f8124b4c80b1f28469ce6f7fa. I'll close this issue therefore.

**Comment by enyo** [comment]
I see. Thanks for the explanation.
